File: Sample Project Obj C/Sample Project/ViewController.m
//
//  ViewController.m
//  Sample Project
//
//  Created by Muhammad Zubair on 5/5/16.
//  Copyright © 2016 MediaBrix. All rights reserved.
//

#import "ViewController.h"

@interface ViewController () <MediaBrixDelegate>

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    [MediaBrix initMediaBrixDelegate:self withBaseURL:@"http://mobile.mediabrix.com/v2/manifest" withAppID:@"TwwvxoFnJn"];
    [self loadAd];

}

-(void)loadAd{
    [[MediaBrix sharedInstance]loadAdWithIdentifier:@"Babel_Rescue" withViewController:self];
}

-(void)showAd {
    [[MediaBrix sharedInstance]showAdWithIdentifier:@"Babel_Rescue" fromViewController:self reloadWhenFinish:NO];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


#pragma mark - MediaBrixDelegate

-(void)mediaBrixStarted {
    [self loadAd];
}
-(void)mediaBrixAdReady:(NSString *)identifier {
    [self showAd];
}

@end

File: IOS/Example/Swift/SampleApp/SampleApp/ViewController.swift
//
//  ViewController.swift
//  SampleApp
//
//  Created by Punnaghai Puviarasu on 4/26/16.
//  Copyright © 2016 Punnaghai Puviarasu. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    var publisherVar:[AnyHashable: Any] = [AnyHashable: Any]()

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        //Initialize mediabix event handler with the base url & the mediabrix provided app_id.
        MediaBrix.initMediaBrixDelegate(self, withBaseURL: "http://mobile.mediabrix.com/v2/manifest", withAppID: "TwwvxoFnJn")
        
        //initialize te publishervariables dictionary
        publisherVar = MediaBrix.userDefaults().defaultAdData()
        
        //load the ad
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    func showAd(){
        MediaBrix.sharedInstance().showAd(withIdentifier: "Babel_Rally", from: self, reloadWhenFinish: false)
    }
    func loadAd() {
        MediaBrix.sharedInstance().loadAd(withIdentifier: "Babel_Rally", adData: publisherVar, with: self)
    }
    
}

extension ViewController: MediaBrixDelegate {
    public func mediaBrixAdReady(_ identifier: String!) {
        self.showAd()
    }
    func mediaBrixStarted() {
        self.loadAd()
    }
}


